onmt/data/SampledDataset.lua

```diff
@@ -0,0 +1,266 @@
+--[[ Data management and batch creation. Handles data created by `preprocess.lua`. ]]
+require 'torchx'
+
+local SampledDataset = torch.class("SampledDataset")
```

Comment:
With changes made in e276661, the batch prep logic diverged more and I don't think there's a room for sharing.